--- bitcoin-22.0/src/support/lockedpool.cpp.orig	2021-09-08 20:16:07.000000000 +0200
+++ bitcoin-22.0/src/support/lockedpool.cpp	2023-07-04 11:29:13.358676698 +0200
@@ -26,6 +26,7 @@
 #include <iomanip>
 #include <iostream>
 #endif
+#include <stdexcept>
 
 LockedPoolManager* LockedPoolManager::_instance = nullptr;
 
--- bitcoin-22.0/src/util/bip32.h.orig	2021-09-08 20:16:07.000000000 +0200
+++ bitcoin-22.0/src/util/bip32.h	2023-07-04 11:31:38.309474568 +0200
@@ -6,6 +6,7 @@
 #define BITCOIN_UTIL_BIP32_H
 
 #include <attributes.h>
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- bitcoin-22.0/src/util/string.h.orig	2021-09-08 20:16:07.000000000 +0200
+++ bitcoin-22.0/src/util/string.h	2023-07-04 11:34:16.117022848 +0200
@@ -9,6 +9,7 @@
 
 #include <algorithm>
 #include <array>
+#include <cstdint>
 #include <cstring>
 #include <locale>
 #include <sstream>
